src/cloud_provider/aws/kubernetes/roles.rs

```diff
@@ -0,0 +1,108 @@
+use self::rusoto_iam::{
+    CreateServiceLinkedRoleError, CreateServiceLinkedRoleRequest, CreateServiceLinkedRoleResponse,
+    GetRoleError, GetRoleRequest, GetRoleResponse, Iam, IamClient,
+};
+use crate::error::{EngineError, SimpleError, SimpleErrorKind};
+use crate::models::Context;
+use futures::TryFutureExt;
+use rusoto_core::{Client, HttpClient, Region, RusotoError};
+use rusoto_credential::StaticProvider;
+use tokio::macros::support::Future;
+use tokio::runtime::Runtime;
+
+extern crate rusoto_iam;
+
+pub struct Role {
+    pub role_name: String,
+    pub service_name: String,
+    pub description: String,
+}
+
+pub fn get_default_roles_to_create() -> Vec<Role> {
+    let mut defaults_role_to_create: Vec<Role> = Vec::new();
+    defaults_role_to_create.push(Role {
+        role_name: "AWSServiceRoleForAmazonElasticsearchService".to_string(),
+        service_name: "es.amazonaws.com".to_string(),
+        description: "role permissions policy allows Amazon ES to complete create, delete, describe,  modify on ec2 and elb".to_string(),
+    });
+    defaults_role_to_create
+}
+
+impl Role {
+    pub fn new(role_name: String, service_name: String, description: String) -> Self {
+        Role {
+            role_name,
+            service_name,
+            description,
+        }
+    }
+
+    pub async fn is_exist(&self, access_key: &str, secret_key: &str) -> Result<bool, SimpleError> {
```

Comment:
Why async?

Reply:
because of rusoto async way, but it's wrapped into func that calls this one